### PR TITLE
JAX docs tutorial: Installation

### LINKED ISSUE #18585: Tracking issue: Documentation update
Started with #18269

The plan is to create an entirely new cohesive documentation structure in `docs/tutorials`, to move sections of the existing docs there, and then to remove the old versions leaving HTTP redirects in their place.

Existing drafts (on the main branch) can be seen at https://jax.readthedocs.io/en/latest/tutorials/: note that this is currently unlisted in the website table of contents.
### Indroductory
- [x] Quickstart @jakevdp #18613
- [x] Key Concepts @jakevdp #20347 
- [x] ~Thinking in JAX @8bitmp3 #18581~ removed in #20802
- [x] Just-in-time compilation @jakevdp #18627
- [x] Automatic vectorization @jakevdp #18626
- [x] Automatic differentiation @8bitmp3 #18676
- [x] Debugging @8bitmp3 #18582
- [x] Pseudorandom numbers @jakevdp #18631
- [x] Working with PyTrees @mtthss #18864
- [x] Introduction to sharded computation @jakevdp #20779
- [x] Stateful computations @jakevdp #20732
- [ ] Example: Writing a simple neural network

### Intermediate
- [x] Parallel computation
- [x] Advanced automatic differentiation @8bitmp3 
- [x] Advanced vectorization (collectives, batched map)
- [x] Gradient checkpointing
- [x] Advanced debugging
- [x] External callbacks @jakevdp #18616
- [ ] ~Profiling and performance @dfm~ *punting on this for now*

### Advanced
- [ ] How JAX primitives work
- [ ] Understanding Jaxprs
- [ ] Writing a custom Jaxpr interpreter
- [ ] Ahead-of-time lowering and computation
- [ ] Custom operations for GPUs with C++ and CUDA
- [ ] Pallas: a JAX kernel language

### Reference
- [x] Installation @8bitmp3 #18580 moved to main page in #20784

### PR DESCRIPTION
**Pull request by 8bitmp3**
@jakevdp 

https://github.com/google/jax/issues/18585

### CODE REVIEW

File: docs/tutorials/installation.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
+# How to install JAX
+
```
Comment (jakevdp) [inline docs/tutorials/installation.md:2]: You previously mentioned the idea of putting the most common commands up top. How about adding something like this here?

> Most jax installation will look something like this:
> ```
> # CPU: Linux/OSX/Windows
> $ pip install "jax[cpu]"
>
> # GPU: Linux with CUDA 12
> pip install "jax[cuda11_pip]" -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html
> ```
> For other platforms and architectures, see the sections below.

File: docs/tutorials/installation.md [hunk 2]
```diff
@@ -1,10 +1,319 @@
...
 
-```{note}
-This is a placeholder for a section in the new {ref}`jax-tutorials`.
+This guide provides instructions for:
 
-For the time being, you may find some related content in the old documentation:
-- {doc}`../installation`
+- Installing JAX binary packages for supported platforms using `pip` or `conda`
+- Using [NVIDIA GPU Docker containers](#docker-containers-nvidia-gpu)
+- [Building JAX from source](developer.md#building-from-source)
```
Comment (jakevdp) [inline docs/tutorials/installation.md:9]: To ensure that we don't inadvertently break links, we should prefer cross-refrences here (i.e. ``` {ref}`label` ``` with the labeled section header specified via `(label)` in the line above the markdown heading).

File: docs/tutorials/installation.md [hunk 3]
```diff
@@ -1,10 +1,319 @@
...
+|------------------|--------------------------|--------------------|----------------------------|----------------------------|-----------------|-----------------------------|
+| CPU              | [yes](#cpu)              | [yes](#cpu)        | [yes](#cpu)                | [yes](#cpu)                | [yes](#cpu)     | [yes](#cpu)                 |
+| NVIDIA GPU       | [yes](#nvidia-gpu)       | [yes](#nvidia-gpu) | no                         | n/a                        | no              | [experimental](#nvidia-gpu) |
+| Google Cloud TPU | [yes](#google-tpu)       | n/a                | n/a                        | n/a                        | n/a             | n/a                         |
+| AMD GPU          | [experimental](#amd-gpu) | no                 | no                         | n/a                        | no              | no                          |
+| Apple GPU        | n/a                      | no                 | [experimental](#apple-gpu) | [experimental](#apple-gpu) | n/a             |  n/a                        |
+
```
Comment (jakevdp) [inline docs/tutorials/installation.md:37]: We could think about replacing links in the table with sphinx cross-references as well, but that might take a bit of work. Probably fine to leave it for later.

File: docs/tutorials/installation.md [hunk 4]
```diff
@@ -1,10 +1,319 @@
...
+|--------------------------------|-----------------------------------------------------------------------------------------------------------------|
+| CPU                            | `pip install -U "jax[cpu]"`                                                                                     |
+| NVIDIA GPU on x86_64 (CUDA 12) | `pip install -U "jax[cuda12_pip]" -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html`        |
+| NVIDIA GPU on x86_64 (CUDA 11) | `pip install -U "jax[cuda11_pip]" -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html` |
+| Google Cloud TPU               | `pip install -U "jax[tpu]" -f https://storage.googleapis.com/jax-releases/libtpu_releases.html`                 |
+| AMD GPU                        | Use [Docker](https://hub.docker.com/r/rocm/jax) or [build from source](https://jax.readthedocs.io/en/latest/developer.html#additional-notes-for-building-a-rocm-jaxlib-for-amd-gpus). |
+| Apple GPU                      | Follow [Apple's instructions](https://developer.apple.com/metal/jax/).
```
Comment (jakevdp) [inline docs/tutorials/installation.md:47]: This seems duplicative with the first paragraph above. Should we move the table there instead?

Comment (8bitmp3) [reply docs/tutorials/installation.md:47]: 100%

File: docs/tutorials/installation.md [hunk 5]
```diff
@@ -21,31 +20,34 @@ $ pip install "jax[cuda11_pip]" -f https://storage.googleapis.com/jax-releases/j
 # Follow Apple's instructions at https://developer.apple.com/metal/jax/
 ```
```
Comment (jakevdp) [inline docs/tutorials/installation.md:21]: I think we can remove this block, because it duplicates what is in the table just below this.

File: docs/tutorials/installation.md [hunk 6]
```diff
@@ -21,31 +20,34 @@ $ pip install "jax[cuda11_pip]" -f https://storage.googleapis.com/jax-releases/j
...
-| AMD GPU          | [experimental](#amd-gpu) | no                 | no                         | n/a                        | no              | no                          |
-| Apple GPU        | n/a                      | no                 | [experimental](#apple-gpu) | [experimental](#apple-gpu) | n/a             |  n/a                        |
-
-Below are just some examples on how to install JAX (do consult with the detailed instructions provided further in this guide):
+|                  | Linux, x86_64            | Linux, aarch64        | macOS, Intel x86_64             | macOS, ARM-based                | Windows, x86_64  | Windows WSL2, x86_64            |
+|------------------|--------------------------|-----------------------|---------------------------------|---------------------------------|------------------|---------------------------------|
+| CPU              | [yes]{ref}`cpu`          | [yes]{ref}`cpu`       | [yes]{ref}`cpu`                 | [yes]{ref}`cpu`                 | [yes]{ref}`cpu`  | [yes]{ref}`cpu`                 |
```
Comment (jakevdp) [inline docs/tutorials/installation.md:43]: Note that the scope of these labels is global to the entire project, so it may be helpful to make them more informative (e.g. `installation-cpu` instead of `cpu`)

**Review by jakevdp — APPROVED** [review]
Looks great! Let's make sure the doc build doesn't error, then we can get this merged. Thanks for all the work on this!
